finality-aleph/src/network.rs

```diff
@@ -87,51 +121,104 @@ impl<B: BlockT, H: ExHashT> Network<B> for Arc<NetworkService<B, H>> {
             log::error!(target: "afa", "remove_set_reserved failed: {}", e);
         }
     }
-}
 
-use rand::{seq::SliceRandom, thread_rng};
+    fn peer_id(&self) -> PeerId {
+        (*self.local_peer_id()).into()
+    }
+}
 
 #[derive(Debug)]
-pub(crate) struct PeerInfo {}
+pub(crate) struct PeerInfo {
+    authentications: HashMap<SessionId, NodeIndex>,
+}
 
 impl PeerInfo {
     fn new() -> Self {
-        PeerInfo {}
+        PeerInfo {
+            authentications: HashMap::new(),
+        }
+    }
+
+    fn authenticated_for(&self, session_id: &SessionId) -> bool {
+        self.authentications.get(session_id).is_some()
+    }
+
+    fn authenticate(&mut self, session_id: SessionId, node_id: NodeIndex) {
+        self.authentications.insert(session_id, node_id);
+    }
+
+    fn iter(&self) -> impl Iterator<Item = (&SessionId, &NodeIndex)> {
+        self.authentications.iter()
     }
 }
 
-#[derive(Debug)]
 struct Peers {
-    pub(crate) peers: HashMap<PeerId, PeerInfo>,
+    all_peers: HashMap<PeerId, PeerInfo>,
+    to_peer: HashMap<SessionId, HashMap<NodeIndex, PeerId>>,
 }
 
 impl Peers {
```

Comment:
Methods of a private struct should be private

Reply:
Huh, right, done.